packages/app-root/src/app/style-registry.js

```diff
@@ -0,0 +1,27 @@
+'use client'
+
+import { useServerInsertedHTML } from 'next/navigation'
+import { ServerStyleSheet, StyleSheetManager } from 'styled-components'
+import { useState } from 'react'
+
+export default function StyledComponentsRegistry({
+  children
+}) {
+  // Only create stylesheet once with lazy initial state
+  // x-ref: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
+  const [styledComponentsStyleSheet] = useState(() => new ServerStyleSheet())
+
+  useServerInsertedHTML(() => {
+    const styles = styledComponentsStyleSheet.getStyleElement()
+    styledComponentsStyleSheet.instance.clearTag()
+    return <>{styles}</>
+  })
+
+  if (typeof window !== 'undefined') return <>{children}</>
+
+  return (
+    <StyleSheetManager sheet={styledComponentsStyleSheet.instance}>
```

Comment:
```suggestion
    <StyleSheetManager disableVendorPrefixes sheet={styledComponentsStyleSheet.instance}>
```

Vendor prefixes are disabled by default in v6, so we may as well disable them in v5 too.